ozaria/site/components/interactive/PageInteractive/draggableOrdering/index.vue

```diff
@@ -136,6 +153,36 @@
 
       resetAnswer () {
         this.promptSlots = this.getShuffledPrompt(this.shuffle)
+
+        this.initializedAnswer = false
+        this.initializeFromPastSubmission()
+      },
+
+      initializeFromPastSubmission () {
+        if (!this.pastCorrectSubmission || this.initializedAnswer) {
+          return
+        }
+
+        this.initializedAnswer = true
+
+        let missingAnswer = false
+        const answer = this.pastCorrectSubmission.submittedSolution.map((elementId) => {
```

Comment:
Because we know the correct solution on the client, do we need to iterate over the user's answers to find it?
This requires us to send over the full list of a user's solutions. Could this data transfer be reduced by checking the completed flag instead and then creating the solution from the interactive data?

Reply:
Yup agreed